src/memory/memoryError.ts

```diff
@@ -0,0 +1,16 @@
+export class MemoryError extends Error {
+  constructor(type: MemoryErrorType, value: number) {
+    super(MemoryError.formatMessage(type, value));
+    this.name = this.constructor.name;
+  }
+
+  private static formatMessage(type: MemoryErrorType, value: number): string {
+    return `type: ${type}, value: ${value}`;
+  }
+}
+
+export enum MemoryErrorType {
```

Comment:
Just wondering, is there a reason for favouring enums rather than extending `MemoryError` and having for example `InvalidOffsetError`, `SegmentationFaultError`, etc?

Reply:
I didn't know how memory errors would play out so I didn't want to create a lot of classes yet